pallets/frequency-tx-payment/src/lib.rs

```diff
@@ -168,12 +171,24 @@ pub mod pallet {
 
 		/// Charge Capacity for transaction payments.
 		type OnChargeCapacityTransaction: OnChargeCapacityTransaction<Self>;
+
+		/// The maxmimum number of capacity calls that can be batched together.
+		#[pallet::constant]
+		type MaximumCapacityBatchLength: Get<u8>;
+
+		type BatchAllProvider: UtilityProvider<OriginFor<Self>, <Self as Config>::RuntimeCall>;
 	}
 
 	#[pallet::event]
 	#[pallet::generate_deposit(pub (super) fn deposit_event)]
 	pub enum Event<T: Config> {}
 
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The maximum amount of requested batched calls was exceeded
+		BatchedCallAmountExceedMaximum,
```

Comment:
Good catch, thank you. I'm partial to `BatchedCallAmountExceedsMaximum` because it states why there are too many batched calls, there is an allowed max